src/main/java/io/rpg/view/popups/TextPopup.java

```diff
@@ -0,0 +1,44 @@
+package io.rpg.view.popups;
+
+import io.rpg.viewmodel.TextPopupViewModel;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Group;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.paint.Color;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class TextPopup extends Scene {
```

Comment:
What do you think of generalising popup concept to separate class (possibly abstract) and just subclassing it for specific popup types? I bring this idea up, because it seems like we could duplicate much less code.

Reply:
TextPopup and TextImagePopup do share most of their code, however other popups that are going to be created will not, maybe besides setting the background image. so I'm not sure if the super class for all of them makes much sense